sdk/appconfiguration/app-configuration/src/appConfigurationClient.ts

```diff
@@ -149,22 +155,25 @@ export class AppConfigurationClient {
     id: ConfigurationSettingId,
     options: GetConfigurationSettingOptions = {}
   ): Promise<GetConfigurationSettingResponse> {
-    return await this.spanner.trace("getConfigurationSetting", options, async (newOptions) => {
+    const opts = operationOptionsToRequestOptionsBase(options);
+    return await this.spanner.trace("getConfigurationSetting", opts, async (newOptions) => {
       const originalResponse = await this.client.getKeyValue(id.key, {
         label: id.label,
         select: newOptions.fields,
         ...newOptions,
-        ...checkAndFormatIfAndIfNoneMatch(id, newOptions)
+        ...checkAndFormatIfAndIfNoneMatch(id, options)
```

Comment:
Shouldnt this be `newOptions`?
```suggestion
        ...checkAndFormatIfAndIfNoneMatch(id, newOptions)
```